src/com/google/enterprise/adaptor/database/ResponseGenerator.java

```diff
@@ -397,44 +403,81 @@ public void generateResponse(ResultSet rs, Response resp)
       overrideDisplayUrl(rs, resp);
 
       ResultSetMetaData rsMetaData = rs.getMetaData();
-      int columnType = Types.OTHER;
-      int numberOfColumns = rsMetaData.getColumnCount();
-      for (int i = 1; i < (numberOfColumns + 1); i++) {
-        String columnName = rsMetaData.getColumnLabel(i);
-        if (columnName.equals(getContentColumnName())) {
-          columnType = rsMetaData.getColumnType(i);
-          log.log(Level.FINEST, "Name: {0}, Type: {1}", new Object[] {columnName, columnType});
-        }
-      }
+      int columnType = rsMetaData.getColumnType(
+          rs.findColumn(getContentColumnName()));
+      log.log(Level.FINEST, "Content column name: {0}, Type: {1}",
+          new Object[] {getContentColumnName(), columnType});
 
       InputStream in = null;
-      Clob clob = null;
-      Blob blob = null;
+      Reader reader = null;
+      Writer writer = null;
       switch (columnType) {
         case Types.VARCHAR:
-          Reader rd = rs.getCharacterStream(getContentColumnName());
-          in = new ByteArrayInputStream(CharStreams.toString(rd).getBytes(Charsets.UTF_8));
+        case Types.LONGVARCHAR:
+          writer = new OutputStreamWriter(resp.getOutputStream(), CHARSET);
+          if (writer != null) {
+            writer.write(rs.getString(getContentColumnName()));
+            writer.close();
```

Comment:
Use try-with-resources to close streams, and note that it's OK for the stream to be null (though you still need to check it in the body before you use it).